### ISSUE TITLE
Graph Editor: nesting `nodegraph` nodes doesn't work

### ISSUE THREAD

**Issue by marwie** [AcademySoftwareFoundation/MaterialX#1446]
Hello,

I'm trying to nest nodegraphs here but it doesn't work and the nested graph is actually lost(?) once I navigate to the parent graph

https://github.com/AcademySoftwareFoundation/MaterialX/assets/5083203/c73809e0-6c37-452d-bde5-4cc3bc49f1f3

after exiting the nested graph and searching only one nodegraph is listed:

![image](https://github.com/AcademySoftwareFoundation/MaterialX/assets/5083203/43888c54-496c-455a-90f8-52d0e1ac836a)

saved mtlx file:
```mtlx
<?xml version="1.0"?>
<materialx version="1.38">
  <nodegraph name="nodegraph1" />
  <nodegraph name="nodegraph2" />
</materialx>
```

when opening the saved nodegraph again it looks *as saved*
![image](https://github.com/AcademySoftwareFoundation/MaterialX/assets/5083203/af1f0d06-2960-4e97-9e3e-706e53701f4c)

**Comment by kwokcb** [comment]
Hi @marwie,
Your hitting a lot of the “good” implementation caveats. :) 
Yes for now MaterialX does not support nested node graphs properly though the editor allows this.

Pinging  @lfl-eholthouser as she’s the owner of the editor for any additional comment on if maybe we should prevent this for now from the UI?

BTW, USD supports this but “flattens” nesting when converting to MaterialX for compatibility in case your workflow requires this.

**Comment by marwie** [comment]
Thanks again for the quick reply.

Knowing this helps a lot already!


---
A quick question on the side:
Where would be a good place to gather some UI feedback to the NodeGraph Editor? The Academy Software Foundation Slack? Or if I would collect them in one issue?

For example: would it be possible to display the current values of connected inputs or outputs
![image](https://github.com/AcademySoftwareFoundation/MaterialX/assets/5083203/7e8340cf-c365-4bee-aebc-88b726cb413f)

**Comment by lfl-eholthouser** [comment]
That's a great point @marwie and thanks for including that additional ui improvement, you're right @kwokcb we should definitely add functionality to prevent this from occurring in the UI.

**Comment by jstone-lucasfilm** [comment]
Thanks for this report, @marwie, and I believe this is now addressed through UI guards in #1503.
